plugins/scenario/hu.bme.mit.gamma.scenario.language/src/hu/bme/mit/gamma/scenario/language/scoping/ScenarioLanguageScopeProvider.xtend

```diff
@@ -63,16 +70,41 @@ class ScenarioLanguageScopeProvider extends AbstractScenarioLanguageScopeProvide
 		}
 	}
 
-	private def IScope getScope(ScenarioDeclaration declaration, EReference reference) {
-		if (reference == ScenarioModelPackage.Literals.SCENARIO_DECLARATION__COMPONENT) {
-			val importedPackages = declaration.imports
+	private def IScope getScope(ScenarioPackage scenarioPackage, EReference reference) {
+		if (reference == ScenarioModelPackage.Literals.SCENARIO_PACKAGE__COMPONENT) {
+			val importedPackages = scenarioPackage.imports
 			return createScopeFor(importedPackages.flatMap[it.components])
 		}
 	}
+	
+	private def IScope getScope(ScenarioCheckExpression check, EReference reference) {
+		if (reference == InterfaceModelPackage.Literals.EVENT_PARAMETER_REFERENCE_EXPRESSION__PORT) {
+			val package = ecoreUtil.getContainerOfType(check, ScenarioPackage)
+			return Scopes.scopeFor(package.component.ports)
+		}
+	}
+
+	private def IScope getScope(EventParameterReferenceExpression expression, EReference reference) {
+		if (reference == InterfaceModelPackage.Literals.EVENT_PARAMETER_REFERENCE_EXPRESSION__PORT) {
+			val package = ecoreUtil.getContainerOfType(expression, ScenarioPackage)
```

Comment:
Does this ID actually work? Isn't `package` a reserved keyword?

Reply:
It works, I think Xtend allows it because it is after the keyword `val`